src/main/java/uk/gov/hmcts/reform/refunds/services/IdamServiceImpl.java

```diff
@@ -162,14 +162,35 @@ public UserIdentityDataDto getUserIdentityData(MultiValueMap<String, String> hea
 
             if (idamUserListResponse != null && !idamUserListResponse.getIdamUserInfoResponseList().isEmpty()) {
 
-                return idamUserListResponse.getIdamUserInfoResponseList().stream()
-//                        .filter(pr -> pr.getRoles().stream().anyMatch(s -> rolePattern.matcher(s).matches()))
-                        .map(IdamUserInfoResponse::getId)
-                        .collect(Collectors.toSet());
+                for (IdamUserInfoResponse idamUserInfoResponse: idamUserListResponse.getIdamUserInfoResponseList()) {
+
+                    userIdentityDataDtoList.add(UserIdentityDataDto.userIdentityDataWith()
+                            .id(idamUserInfoResponse.getId())
+                            .emailId(idamUserInfoResponse.getEmail())
+                            .fullName(idamUserInfoResponse.getForename() + " " + idamUserInfoResponse.getSurname())
+                            .build());
+                }
+
+               return userIdentityDataDtoList;
             }
         }
 
         LOG.error(USER_DETAILS_NOT_FOUND_ERROR_MSG);
         throw new UserNotFoundException(USER_DETAILS_NOT_FOUND_ERROR_MSG);
     }
+
+    private StringBuffer getRoles(List<String> roles) {
+        StringBuffer rolesValue = new StringBuffer("(");
+        if (!roles.isEmpty()) {
+            if (roles.size() > MIN_VALUE) {
```

Comment:
This condition is not required as you are anyway removing the last OR at line 189. So it doesn't matter if it is single role or multiple roles

Reply:
Done